lib/cppcheck.cpp

```diff
@@ -800,6 +800,7 @@ unsigned int CppCheck::checkFile(const std::string& filename, const std::string
             return 0;
         }
 
+#ifdef HAVE_RULES
```

Comment:
I don't know what is right or wrong here. I believe it's often considered bad to use `#ifdef` these should be avoided.
I don't have a strong opinion about these. I don't want to have tons of ifdefs. But a small number of ifdefs with limited scope does not really hurt much.
@chrchr-github @pfultz2 what do you think?

Reply:
I just moved them so there no longer is any reference to the `Settings::rules` without PCRE. We also disabled other conditional settings.

Reply:
ok sure. I think it's fine as long as we just have "few" ifdefs..